### PR TITLE
Added Exploded feature

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Ensured test coverage is 100%

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
   "require": {
     "php": ">=7.3",
     "ext-json": "*",
-    "elegant-bro/interfaces": "^1.0"
+    "elegant-bro/interfaces": "^1.0",
+    "elegant-bro/stringify": "^0.3.1"
```
Comment (stepanets) [inline composer.json]: we have to avoid circular dependencies between libs

File: src/Exploded.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+     * @var Stringify stringify to explode
+     */
+    private $stringify;
+
+    public static function byComma(Stringify $stringify): Exploded
+    {
+        return new static(new \ElegantBro\Stringify\Just(","), $stringify);
+    }
+
+    public function __construct(Stringify $glue, Stringify $stringify)
```
Comment (stepanets) [inline src/Exploded.php]: `glue` is for implode operation, for explode it should be `separator`

File: src/Exploded.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private $glue;
+
+    /**
+     * @var Stringify stringify to explode
+     */
+    private $stringify;
+
+    public static function byComma(Stringify $stringify): Exploded
+    {
+        return new static(new \ElegantBro\Stringify\Just(","), $stringify);
```
Comment (stepanets) [inline src/Exploded.php]: Looks like we need to have `Just` implementations for every type in `Interfaces` library .  You can do without `byComma` constructor at this time and do not import `Stringify\Just`

File: tests/ExplodedTest.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+{
+    /**
+     * @throws Exception
+     */
+    public function testAsArray(): void
+    {
+        $this->assertEquals(
+            ['foo','bar'],
+            (new Exploded(
+                new Just('-'),
```
Comment (stepanets) [inline tests/ExplodedTest.php]: You can use `StubStringify` in tests

**Review by stepanets — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (maxonfjvipon) [reply composer.json]: Done!

File: src/Exploded.php (hunk 2 above)
Comment (maxonfjvipon) [reply src/Exploded.php]: Done!

File: src/Exploded.php (hunk 3 above)
Comment (maxonfjvipon) [reply src/Exploded.php]: Done!

File: tests/ExplodedTest.php (hunk 4 above)
Comment (maxonfjvipon) [reply tests/ExplodedTest.php]: Done!

File: tests/ExplodedTest.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+{
+    /**
+     * @throws Exception
+     */
+    public function testAsArray(): void
+    {
+        $this->assertEquals(
+            ['foo','bar'],
+            (new Exploded(
+                new Stub\StubStringify("-"),
```
Comment (stepanets) [inline tests/ExplodedTest.php]: `Stub\` prefix is redundant

**Review by stepanets — CHANGES_REQUESTED** [review]
And you should rollback composer.lock

File: tests/ExplodedTest.php (hunk 5 above)
Comment (maxonfjvipon) [reply tests/ExplodedTest.php]: Done!

**Review by stepanets — APPROVED** [review]

**Pull request merged**